diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -89,42 +95,82 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
+            isFixedSmthInPreviousStep = false
+
+            if (node.elementType == KtFileElementType.INSTANCE) {
+                node.getChildren(null).forEach {
+                    if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
+                        val isFixedSmthInChildNode = checkLength(it, configuration)
+
+                        if (!isFixedSmthInPreviousStep && isFixedSmthInChildNode) {
+                            isFixedSmthInPreviousStep = true
+                        }
+                    }
                 }
             }
-        }
+        } while (isFixedSmthInPreviousStep)
     }
 
-    @Suppress("UnsafeCallOnNullableType", "TOO_LONG_FUNCTION")
-    private fun checkLength(node: ASTNode, configuration: LineLengthConfiguration) {
+    @Suppress(
+        "UnsafeCallOnNullableType",
+        "TOO_LONG_FUNCTION",
+        "FUNCTION_BOOLEAN_PREFIX"
+    )
+    private fun checkLength(node: ASTNode, configuration: LineLengthConfiguration): Boolean {
+        var isFixedSmthInChildNode = false
+
         var offset = 0
         node.text.lines().forEach { line ->
             if (line.length > configuration.lineLength) {
                 val newNode = node.psi.findElementAt(offset + configuration.lineLength.toInt() - 1)!!.node
+
                 if ((newNode.elementType != TEXT && newNode.elementType != MARKDOWN_INLINE_LINK) || !isKdocValid(newNode)) {
                     positionByOffset = node.treeParent.calculateLineColByOffset()
+
                     val fixableType = isFixable(newNode, configuration)
+
                     LONG_LINE.warnOnlyOrWarnAndFix(
                         configRules, emitWarn,
                         "max line length ${configuration.lineLength}, but was ${line.length}",
                         offset + node.startOffset, node,
                         shouldBeAutoCorrected = fixableType !is None,
                         isFixMode,
                     ) {
-                        // we should keep in mind, that in the course of fixing we change the offset
+                        val textBeforeFix = node.text
                         val textLenBeforeFix = node.textLength
+                        val blankLinesBeforeFix = node.text.lines().size - countCodeLines(node)
+
                         fixableType.fix()
+
+                        val textAfterFix = node.text
                         val textLenAfterFix = node.textLength
-                        // offset for all next nodes changed to this delta
-                        offset += (textLenAfterFix - textLenBeforeFix)
+                        val blankLinesAfterFix = node.text.lines().size - countCodeLines(node)
+
+                        isFixedSmthInChildNode = fixableType !is None
+
+                        if (textBeforeFix == textAfterFix) {
+                            isFixedSmthInChildNode = false
+                        }
+
+                        if (blankLinesAfterFix > blankLinesBeforeFix) {
```

Comment:
please add comment, why we need to unfix in this case

Reply:
Done